File: Data Structures & Algorithms/search-2d-matrix/submission-0.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[0].length; j++) {
                if (i + 1 < matrix.length && target >= matrix[i + 1][0]) {
                    continue;
                }
                if (matrix[i][j] == target) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        nums.sort(reverse=True)
        res = []
        self.combinationSumH(nums, target, [], res)
        return res
    
    def combinationSumH(self, nums, target, cur, res):
        if target == 0:
            res.append(cur)
        elif nums:
            self.combinationSumH(nums[1:], target, cur, res)
            if target - nums[0] >= 0:
                self.combinationSumH(nums, target - nums[0], cur + [nums[0]], res)
        